client/src/actions/smells.js

```diff
@@ -0,0 +1,69 @@
+import { REQUEST_CODESMELLS, RECEIVE_CODESMELLS, ADD_CODESMELLS, ADD_DEFAULT_CODESMELLS } from '../constants/ActionTypes';
+import request from 'superagent';
+
+function requestCodeSmells() {
+
+    return {
+        type: REQUEST_CODESMELLS,
+    }
+
+}
+
+function receiveCodeSmells(json) {
+
+    return {
+        type: RECEIVE_CODESMELLS,
+        payload: {
+        	smells: json
+        }
+    }
+
+}
+
+function addCodeSmellsRequest() {
+    return {
+        type: ADD_CODESMELLS_REQUEST
+    }
+}
+
+// TODO Update depending on where we store the codesmells - DB or config
+export function fetchCodeSmells() {
+	return function (dispatch) {
+		dispatch(requestCodeSmells());
+		return request
+			.get('http://localhost:8000/app/codesmells/')
+			.end(function(err, res) {
+				if(res && res.status === 404) {
+                    reject();
+                }
+                else {
+                    
+                    var data = JSON.parse(res.text);
+                    dispatch(receiveCodeSmells(data));
+                
+                }
+			});
+	}
+}
+
+export function addCodeSmells(userid, codeid, codesmells, redirect="/") {
```

Comment:
Why is there a redirect if you're not using it? Did you copy it from the auth.js file?